test/index.test.js

```diff
@@ -821,6 +821,30 @@ describe('Optimizely', function() {
           revenue: 999
         }]);
       });
+
+      describe('the Optimizely X Fullstack JavaScript client is present', function() {
+        beforeEach(function() {
+          window.optimizelyClientInstance = {};
+          analytics.stub(window.optimizelyClientInstance, 'track');
+        });
+
+        afterEach(function() {
+          window.optimizelyClientInstance.track.restore();
+        });
+
+        // @TODO: please enable once we fix `track.userId()`, which at the moment returns undefined
```

Comment:
@mikeng13 Ahh yeah this is a known problem in our test suite actually.

@tsholmes Is this something we can fix? When we call `.identify()` the userId is not accessible inside a track function... I've ran into this before. Any workaround here? I really don't want us to use `this.analytics.user().id()` in the integration code...